troubleshooting/rocksdb/src/solution/java/com/ververica/flink/training/solutions/RocksDBTuningJobOptionsFactory.java

```diff
@@ -0,0 +1,133 @@
+ * https://github.com/facebook/rocksdb/wiki/RocksDB-Tuning-Guide.
+ */
+@SuppressWarnings("unused")
+public class RocksDBTuningJobOptionsFactory implements ConfigurableRocksDBOptionsFactory {
+    public static final org.slf4j.Logger LOG =
+            LoggerFactory.getLogger(RocksDBTuningJobOptionsFactory.class);
+
+    private static final long serialVersionUID = 1L;
+
+    private static final ConfigOption<Boolean> JAVA_LOGGING =
+            key("state.backend.rocksdb.custom.javalogging")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription("Redirect RocksDB logging to Java (max reduce performance)");
+
+    private static final ConfigOption<Duration> DUMP_STATS_INTERVAL =
+            key("state.backend.rocksdb.custom.dumpstats")
+                    .durationType()
+                    .defaultValue(Duration.ofMinutes(5))
+                    .withDescription("If not zero, dump RocksDB stats this often (in seconds)");
+
+    private static final ConfigOption<CompressionType> COMPRESSION =
+            key("state.backend.rocksdb.custom.compression")
+                    .enumType(CompressionType.class)
+                    .noDefaultValue()
+                    .withDescription(
+                            Description.builder()
+                                    .text("Configures RocksDB compression")
+                                    .linebreak()
+                                    .text(
+                                            "For more information, please refer to %s",
+                                            link(
+                                                    "https://github.com/facebook/rocksdb/wiki/Compression"))
+                                    .build());
+
+    public static final ConfigOption<Integer> MAX_BACKGROUND_JOBS =
+            key("state.backend.rocksdb.custom.maxjobs")
+                    .intType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Specifies the maximum number of concurrent background jobs (both flushes and compactions combined). "
+                                    + "RocksDB has default configuration as '2'.");
+
+    private final DefaultConfigurableOptionsFactory defaultFactory =
+            new DefaultConfigurableOptionsFactory();
+    private ReadableConfig configuration;
+
+    @Override
+    public DBOptions createDBOptions(
+            DBOptions currentOptions, Collection<AutoCloseable> handlesToClose) {
+
+        defaultFactory.setLogLevel(InfoLogLevel.INFO_LEVEL);
+        defaultFactory.setLogFileNum(10);
+        defaultFactory.setMaxLogFileSize("10MB");
+        currentOptions = defaultFactory.createDBOptions(currentOptions, handlesToClose);
+
+        Duration statsDumpPeriod = configuration.get(DUMP_STATS_INTERVAL);
+        currentOptions.setStatsDumpPeriodSec((int) statsDumpPeriod.toMillis());
```

Comment:
You are passing ms to a parameter that expects seconds. With the default 5 minute duration, you are setting this to dump stats every 3.5 days 😱  There also needs to be some rounding logic but we can probably keep it simple for the training exercise. 

```suggestion
        currentOptions.setStatsDumpPeriodSec(Math.max(1, (int) statsDumpPeriod.toMillis() / 1000));
```